projects/iio_demo_freeRTOS/src/platform/maxim/main.c

```diff
@@ -0,0 +1,184 @@
+ ********************************************************************************
+* Copyright 2023(c) Analog Devices, Inc.
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*  - Redistributions of source code must retain the above copyright
+*	notice, this list of conditions and the following disclaimer.
+*  - Redistributions in binary form must reproduce the above copyright
+*	notice, this list of conditions and the following disclaimer in
+*	the documentation and/or other materials provided with the
+*	distribution.
+*  - Neither the name of Analog Devices, Inc. nor the names of its
+*	contributors may be used to endorse or promote products derived
+*	from this software without specific prior written permission.
+*  - The use of this software may or may not infringe the patent rights
+*	of one or more patent holders.  This license does not release you
+*	from the requirement that you obtain separate licenses from these
+*	patent holders to use this software.
+*  - Use of the software either in source or binary form, must be run
+*	on or directly connected to an Analog Devices Inc. component.
+*
+* THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+* IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+* MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+* IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+* LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "platform_includes.h"
+#include "common_data.h"
+
+#ifdef IIO_EXAMPLE
+#include "iio_example.h"
+#endif
+
+#ifdef IIO_TRIGGER_EXAMPLE
+#include "iio_trigger_example.h"
+#endif
+
+/* Enables/disables tick-less mode */
+unsigned int disable_tickless = 1;
+
+/* Stringification macros */
+#define STRING(x) STRING_(x)
+#define STRING_(x) #x
+
+/* Console ISR selection */
+#if (CONSOLE_UART == 0)
+#define UARTx_IRQHandler UART0_IRQHandler
+#define UARTx_IRQn UART0_IRQn
+
+#else
+#error "Please update ISR macro for UART CONSOLE_UART"
+#endif
+
+/**
+ * @brief iiod example task
+*/
+
+void iiodTask(void *pvParameters)
+{
+	iio_example_main();
```

Comment:
I think for this example it's ok to have one blinking LED task and one IIO task. Two iiod's don't really make sense as they can't really work together with one UART.